pkg/manifests/manifests.go

```diff
@@ -1424,6 +1425,29 @@ func (f *Factory) PrometheusK8s(grpcTLS *v1.Secret, telemetrySecret *v1.Secret)
 	for k := range p.Spec.RemoteWrite {
 		rw := &p.Spec.RemoteWrite[k]
 
+		// Fixes: https://issues.redhat.com/browse/OCPBUGS-38289
+		// The code is condensed here to simplify backporting.
+		// This is a temporary workaround because the NoProxy field isn't exported on the Prometheus CRD in older versions.
+		// If the cluster proxy or remoteWrite config change, this code will re-run, ensuring no stale state.
+		// The httpproxy package used here is the same as used by Prometheus, which helps with consistency.
+		if f.proxy.HTTPSProxy() == "" && f.proxy.HTTPProxy() == "" {
+			continue
+		}
+		// Check if the endpoint URL needs proxying.
+		proxyConfig := &httpproxy.Config{
+			HTTPProxy:  f.proxy.HTTPProxy(),
+			HTTPSProxy: f.proxy.HTTPSProxy(),
+			NoProxy:    f.proxy.NoProxy(),
+		}
+		proxyFunc := proxyConfig.ProxyFunc()
+		// Ignore errors here.
+		u, _ := url.Parse(rw.URL)
+		proxyURL, err := proxyFunc(u)
+		if err == nil && proxyURL == nil {
+			// Proxing will not be used, no need to set it.
+			continue
+		}
+
 		if f.proxy.HTTPProxy() != "" {
```

Comment:
hmm shouldn't we remove this code and use `proxyURL` instead?

Reply:
I adjust the code to make it simulate what would happen if CMO set/overwrites the proxy URL here.